telegram/message.py

```diff
@@ -338,10 +338,14 @@ def chat_id(self):
 
     @property
     def link(self):
-        """:obj:`str`: Convenience property. If the chat of the message is a supergroup or a
-        channel and has a :attr:`Chat.username`, returns a t.me link of the message."""
-        if self.chat.type in (Chat.SUPERGROUP, Chat.CHANNEL) and self.chat.username:
-            return "https://t.me/{}/{}".format(self.chat.username, self.message_id)
+        """:obj:`str`: Convenience property. If the chat of the message is not
+        a private chat, returns a t.me link of the message."""
+        if self.chat.type is not Chat.PRIVATE:
+            if self.chat.username:
+                to_link = self.chat.username
+            else:
+                to_link = "c/{}".format(self.chat.id)
```

Comment:
As far as my testing goes, this should be
 ```
to_link = "c/{}".format(abs(int(self.chat.id)))
```
The `int` just in case, `chat.id` is a string and the `abs` because Telegram redirects to telegram.org, if you insert the negative chat id of groups.

Reply:
actually, we have to remove the -100 for groups.